### PR TITLE
Realized function getNumPairs

### PR DESCRIPTION
**Pull request by Renuarqf**

### CODE REVIEW

**Review by kobuqa — COMMENTED** [review]
could you please avoid O(n^2) time complexity ?:)

File: index.js [hunk 1]
```diff
@@ -11,17 +11,19 @@
  *     getNumPairs([2,3,4,5], 5, 7); // Return 4;
  */
 function getNumPairs(arr, l, r) {
-  const sorted = arr.sort((a, b) => a - b);
   let pairs = [];
-  for (let i = 0; i < sorted.length; i++) {
-    for (let j = i + 1; j < sorted.length; j++) {
-      const sum = sorted[i] + sorted[j];
+  for (let i = 0; i < arr.length; i++) {
```
Comment (kobuqa) [inline index.js:15]: ~~sort - O(n)~~ - Ok. you've removed that step
nested loops - O(n^2) - Not ok,  here is a key problem
filter - O(n) - Ok

File: index.js [hunk 2]
```diff
@@ -11,7 +11,17 @@
  *     getNumPairs([2,3,4,5], 5, 7); // Return 4;
  */
 function getNumPairs(arr, l, r) {
-  // your implementation here
+  const sorted = arr.sort((a, b) => a - b);
+  let pairs = [];
+  for (let i = 0; i < sorted.length; i++) {
+    for (let j = i + 1; j < sorted.length; j++) {
```
[the remark this answers is not in the document]
Comment (Renuarqf) [reply index.js]: Removed sorting

**Review by kobuqa — DISMISSED** [review]
in worst case it's still O(n^2)

File: index.js (hunk 1 above)
Comment (Renuarqf) [reply index.js:15]: Is not possible exclude nested loops  to create a pairs of array

**Pull request closed without merging**